tests/test_api.py

```diff
@@ -126,3 +127,17 @@ def test_export(self):
         processor = api.GradeCSVProcessor(course_id=self.course_id)
         rows = list(processor.get_iterator())
         assert len(rows) == 4
+
+    @patch('lms.djangoapps.grades.api.graded_subsections_for_course_id')
+    def test_subsection_max_min(self, mock_graded_subsections):
+        self._make_enrollments()
+        subsection = MagicMock()
+        subsection.location = MagicMock()
+        subsection.display_name = 'asdf'
+        subsection.location.block_id = self.block_id_in_module_id
+        mock_graded_subsections.return_value = [subsection]
+        # should filter out everything; all grades are 1 from mock_apps grades api
+        processor = api.GradeCSVProcessor(course_id=self.course_id, subsection=self.block_id, subsection_grade_min=2)
```

Comment:
Could you add a case that tests the max condition as well? Maybe `subsection_grade_min=0, subsection_grade_max=0`?

Reply:
I could. What value is there to that, though? I hate needing to mix ddt and mock.patch; I feel like the order of extra parameters passed to the test is really counterintuitive

Reply:
Passing in `subsection_grade_min=2` hits the first condition of the if statement and doesn't execute the condition checking against `subsection_grade_max`.

```
if effective_grade < self.subsection_grade_min or (
    self.subsection_grade_max and effective_grade > self.subsection_grade_max):
```

I was thinking that, for completeness, passing in `subsection_grade_min=0`, `subsection_grade_max=0` wouldn't cause this short circuiting and would test the `subsection_grade_max` logic as well. I tend to prefer to test all paths in unit tests.

You could initialize a new `GradeCSVProcessor`. I know it's not as DRY as it could be using ddt, but I feel like for two test conditions, it's fine. I don't expect we'd need to continue that pattern within this test. What do you think?

Reply:
Yeah, you're write that the extra messiness in this test isn't really much duplication, and it's dry enough as you described it. Thanks for talking through the trade-offs with me